lib/fluent/plugin/checkpoint_metadata_cache.rb

```diff
@@ -0,0 +1,111 @@
+#
+# Fluentd Kubernetes Metadata Filter Plugin - Enrich Fluentd events with
+# Kubernetes metadata
+#
+# Copyright 2017 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require "sqlite3"
+
+module KubernetesMetadata
+  module CheckPointCache
```

Comment:
Why are we calling this `CheckPointCache`instead of something more meaningful IMO, like `PersistentCache` or `PersistentLruCache`.  This is what this feature adds correct?  A cache that survives restarts of fluentd and rolls off old entries that have timed out.

Reply:
Agreed, changed the name to something more meaningful.